fmriprep/config.py

```diff
@@ -401,12 +404,14 @@ def init(cls):
         if cls._layout is None:
             import re
             from bids.layout import BIDSLayout
-            work_dir = cls.work_dir / 'bids.db'
-            work_dir.mkdir(exist_ok=True, parents=True)
+            _db_path = cls.database_path or (cls.work_dir / cls.run_uuid / 'bids.db')
+            _db_path.mkdir(exist_ok=True, parents=True)
+            _reset_db = bool(cls.database_path is not None)
```

Comment:
This will reindex the dataset if `--database_path` is provided, correct? What is the need for this?

Reply:
I think this is a mistake, the indexing should only be reset if database_path was not provided.